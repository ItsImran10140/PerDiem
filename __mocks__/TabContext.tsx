import React from 'react';

const mockSetActiveTab = jest.fn();
const mockActiveTab = 'catch';

export const TabProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return (
    <div data-testid="tab-provider">
      {children}
    </div>
  );
};

export const useTabContext = () => ({
  activeTab: mockActiveTab,
  setActiveTab: mockSetActiveTab,
});

// Export the mock functions for test assertions
export const tabContextMocks = {
  mockSetActiveTab,
  mockActiveTab,
}; 