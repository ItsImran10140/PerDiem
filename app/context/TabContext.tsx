import React, { createContext, useState, useContext, useEffect } from 'react';
import AsyncStorage from '@react-native-async-storage/async-storage';

type TabType = 'catch' | 'team' | 'settings';

interface TabContextType {
  activeTab: TabType;
  setActiveTab: (tab: TabType) => void;
}

const TabContext = createContext<TabContextType | undefined>(undefined);

export const TabProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const [activeTab, setActiveTabState] = useState<TabType>('catch');

  useEffect(() => {
    const loadTabState = async () => {
      try {
        const savedTab = await AsyncStorage.getItem('activeTab');
        if (savedTab) {
          setActiveTabState(savedTab as TabType);
        }
      } catch (error) {
        console.error('Error loading tab state:', error);
      }
    };

    loadTabState();
  }, []);


  const setActiveTab = (tab: TabType) => {
    setActiveTabState(tab);
    
  
    AsyncStorage.setItem('activeTab', tab).catch(error => {
      console.error('Error saving tab state:', error);
    });
  };

  return (
    <TabContext.Provider value={{ activeTab, setActiveTab }}>
      {children}
    </TabContext.Provider>
  );
};

export const useTabContext = () => {
  const context = useContext(TabContext);
  if (context === undefined) {
    throw new Error('useTabContext must be used within a TabProvider');
  }
  return context;
}; 