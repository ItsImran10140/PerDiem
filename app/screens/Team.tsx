import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { NavigationProp } from '@react-navigation/native';
import TabBar from '../components/TabBar';

interface RouterProps {
  navigation: NavigationProp<any, any>;
}

const Team = ({ navigation }: RouterProps) => {
  return (
    <View style={styles.container}>
      <View style={styles.content}>
        <Text style={styles.title}>My Team</Text>
        <Text style={styles.subtitle}>Your captured Pokemon will appear here</Text>
      </View>
      <View style={styles.tabBarContainer}>
        <TabBar navigation={navigation} />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "white",
    position: 'relative',
  },
  content: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20,
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    color: '#686D76',
    marginBottom: 10,
  },
  subtitle: {
    fontSize: 16,
    color: '#686D76',
    textAlign: 'center',
  },
  tabBarContainer: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
  }
});

export default Team; 