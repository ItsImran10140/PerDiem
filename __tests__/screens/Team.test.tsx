import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { View } from 'react-native';
import Team from '../../app/screens/Team';
import { TabProvider } from '../../app/context/TabContext';
import { NavigationProp } from '@react-navigation/native';


jest.mock('../../app/components/TabBar', () => {
  const MockedTabBar = () => null;
  return MockedTabBar;
});


jest.mock('../../app/context/TabContext', () => ({
  TabProvider: ({ children }: { children: React.ReactNode }) => children,
  useTabContext: () => ({
    activeTab: 'team',
    setActiveTab: jest.fn(),
  }),
}));


const mockNavigation = {
  navigate: jest.fn(),
  goBack: jest.fn(),
  setOptions: jest.fn(),
} as unknown as NavigationProp<any, any>;

describe('Team Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { getByText } = render(
      <TabProvider>
        <Team navigation={mockNavigation} />
      </TabProvider>
    );
    
 
    expect(getByText('My Team')).toBeTruthy();
    
  
    expect(getByText('Your captured Pokemon will appear here')).toBeTruthy();
  });

  it('has the correct layout structure', () => {
    const { UNSAFE_getAllByType } = render(
      <TabProvider>
        <Team navigation={mockNavigation} />
      </TabProvider>
    );
    
 
    const viewComponents = UNSAFE_getAllByType(View);
    expect(viewComponents.length).toBeGreaterThan(0);
  });

  it('renders with the correct structure', () => {
    const { toJSON } = render(
      <TabProvider>
        <Team navigation={mockNavigation} />
      </TabProvider>
    );
    const tree = toJSON();
    

    expect(tree).toMatchSnapshot();
  });

  it('passes navigation prop to TabBar', () => {
    const { UNSAFE_getAllByProps } = render(
      <TabProvider>
        <Team navigation={mockNavigation} />
      </TabProvider>
    );
    
    const tabBars = UNSAFE_getAllByProps({ navigation: mockNavigation });
    expect(tabBars.length).toBeGreaterThan(0);
    expect(tabBars[0].props.navigation).toBe(mockNavigation);
  });

  it('has the correct styling', () => {
    const { toJSON } = render(
      <TabProvider>
        <Team navigation={mockNavigation} />
      </TabProvider>
    );
    const tree = toJSON();
    
    expect(tree.props.style).toEqual(expect.objectContaining({
      flex: 1,
      backgroundColor: "white",
      position: 'relative',
    }));
  });
}); 