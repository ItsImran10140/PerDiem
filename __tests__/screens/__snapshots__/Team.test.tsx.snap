// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team Component renders with the correct structure 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "position": "relative",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#686D76",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginBottom": 10,
        }
      }
    >
      My Team
    </Text>
    <Text
      style={
        {
          "color": "#686D76",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Your captured Pokemon will appear here
    </Text>
  </View>
  <View
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
      }
    }
  />
</View>
`;
